web/src/components/EnableSimulatorButton.tsx

```diff
@@ -1,69 +1,136 @@
+  SwitchProps,
+  Text,
+  Tooltip,
   useColorModeValue,
 } from "@chakra-ui/react";
 import * as React from "react";
-import { useRecoilValue, useSetRecoilState } from "recoil";
+import { useRecoilState, useRecoilValue } from "recoil";
 
 import { trackAnalyticsEvent } from "@/analytics";
 import { setSessionController } from "@/data/queries";
 import { connectionConfig, simulatorEnabled } from "@/data/recoil";
-import { useConnectionState, useMountedCallback } from "@/view/hooks/hooks";
+import { useConnectionState } from "@/view/hooks/hooks";
 import { useSession } from "@/view/hooks/useSession";
 
-export const EnableSimulatorButton = () => {
-  const setEnabled = useSetRecoilState(simulatorEnabled);
+export const EnableSimulatorWarning = () => {
+  return (
+    <Alert status="warning" borderRadius="md">
+      <AlertIcon />
+      <Box display="flex" flexDirection="column">
+        <AlertTitle>The simulator is disabled</AlertTitle>
+        <AlertDescription>
+          This application uses a simulator to generate new data like live
+          notifications and subscribers. To experience the full power of
+          real-time digital marketing, please enable the simulator in the nav
+          bar.
+        </AlertDescription>
+      </Box>
+    </Alert>
+  );
+};
+
+export const SimulatorToggler = ({
+  switchProps,
+  containerProps,
+}: {
+  switchProps?: SwitchProps;
+  containerProps?: FormControlProps;
+}) => {
+  const [enabled, setEnabled] = useRecoilState(simulatorEnabled);
   const { session, refresh: refreshSession } = useSession();
   const config = useRecoilValue(connectionConfig);
   const { connected, initialized } = useConnectionState();
+  const [toggling, setToggling] = React.useState(false);
 
-  const [enabling, enablingCtrl] = useBoolean(false);
-  const stopSpinner = useMountedCallback(
-    () => enablingCtrl.off,
-    [enablingCtrl]
-  );
-  const onEnableSimulator = React.useCallback(async () => {
-    enablingCtrl.on();
-    trackAnalyticsEvent("enable-simulator");
-
+  const onToggleSimulator = React.useCallback(async () => {
+    setToggling(true);
+    const newState = !enabled;
+    trackAnalyticsEvent("change-simulator-state", {
+      enabled: newState.toString(),
+    });
     if (connected && initialized) {
-      await setSessionController(config, session.sessionID, true);
+      await setSessionController(config, session.sessionID, newState);
     }
-    setEnabled(true);
-
+    setEnabled(newState);
     refreshSession();
-    stopSpinner();
+    setToggling(false);
   }, [
     config,
     connected,
-    enablingCtrl,
     initialized,
-    refreshSession,
     session.sessionID,
+    enabled,
+    refreshSession,
     setEnabled,
-    stopSpinner,
   ]);
 
   return (
-    <Alert status="warning" borderRadius="md">
-      <AlertIcon />
-      <AlertTitle>The simulator is disabled</AlertTitle>
+    <FormControl {...containerProps} gap={3}>
+      <FormLabel
+        htmlFor="simulatorSwitch"
```

Comment:
@sandeshvijayraj Exactly, because it's an identifier. Usually camel case is used for variables and pascal case for components. Thanks for changing it!